### PR TITLE
New: adds full compatibility with mPDF to the Simple Template

### LINKED ISSUE #773: CSS styles missing with mPDF active
With the mPDF plugin active, some of the CSS styling is missing for all documents including `tr` borders and most notably, the footer width. The footer issue does not exist with the invoice though as it

Invoice without mPDF active

Screenshot 2024-04-19 081230

Invoice with mPDF active

Screenshot 2024-04-19 081433

A proforma with mPDF active

Screenshot 2024-04-19 081457

It's the same for the other other documents except the invoice. The invoice has additional inline styles that the other documents don't have

Invoice

Screenshot 2024-04-19 081846

Proforma

Screenshot 2024-04-19 081823

### PR DESCRIPTION
**Pull request by alexmigf**
closes #773 

This PR aims to achieve full mPDF compatibility for the Simple template, ultimately eliminating the need for mPDF templates altogether. If successful, we can extend this approach to the Premium Templates as well.

Requires this PR branch for testing: https://github.com/wpovernight/woocommerce-pdf-ips-mpdf/pull/21

### CODE REVIEW

**Review by MohamadNateqi — CHANGES_REQUESTED** [review]

**Review by dwalkerpriv — CHANGES_REQUESTED** [review]
The documents load as usual. However, the styling issue with the `ul` and `tr` tags still is there

mpdf enabled

Would it be better to use a tag different from the `ul` for the meta list or swap the `li` tags for `p` tags?

I've tried applying the `border-bottom` style to the `tr` tags directly but they don't get rendered, even the [doc here](https://mpdf.github.io/css-stylesheets/supported-css.html) says that these styles are compatible on table elements.

File: templates/Simple/style.css [hunk 1]
```diff
@@ -164,19 +175,23 @@ table.order-details {
 	width: 20%;
 }
 
-.order-details tr {
+.order-details tr,
+.notes-totals tr {
 	page-break-inside: always;
```
[the remark this answers is not in the document]
Comment (alexmigf) [reply templates/Simple/style.css:180]: It's still supported by Dompdf. We could address this in the future in another PR.

File: templates/Simple/invoice.php [hunk 2]
```diff
@@ -106,45 +105,51 @@
 	</thead>
 	<tbody>
 		<?php foreach ( $this->get_order_items() as $item_id => $item ) : ?>
-			<tr class="<?php echo apply_filters( 'wpo_wcpdf_item_row_class', 'item-'.$item_id, esc_attr( $this->get_type() ), $this->order, $item_id ); ?>">
+			<tr class="<?php echo apply_filters( 'wpo_wcpdf_item_row_class', 'item-' . $item_id, esc_attr( $this->get_type() ), $this->order, $item_id ); ?>">
 				<td class="product">
 					<?php $description_label = __( 'Description', 'woocommerce-pdf-invoices-packing-slips' ); // registering alternate label translation ?>
-					<span class="item-name"><?php echo $item['name']; ?></span>
+					<p class="item-name"><?php echo $item['name']; ?></p>
```
[the remark this answers is not in the document]
Comment (alexmigf) [reply templates/Simple/invoice.php:111]: Since `<p>` is a block-level element, there should not be any text preceding the item name.

**Comment by alexmigf** [comment]
@dwalkerpriv fixed

**Review by MohamadNateqi — APPROVED** [review]
LGTM!

I think we should have an update on the batch number plugin.
Removing `.wc-item-meta p` from the style, made it two lines:

image

Since it's using the old format:

image

**Comment by alexmigf** [comment]
@MohamadNateqi I'm not seeing a `<p>` there inside the list.

**Comment by MohamadNateqi** [comment]
@alexmigf Sorry the second image was wrong. This is the correct one:

image

That image was for the time the mpdf plugin was activated.

**Review by dwalkerpriv — APPROVED** [review]
Works well on my end now 👍

**Comment by alexmigf** [comment]
@MohamadNateqi I'm not seeing any `<p>` tag currently:

Screenshot from 2024-06-05 09-39-35

Also, mPDF extension has some code to remove the `<p>` tags here: https://github.com/wpovernight/woocommerce-pdf-ips-mpdf/blob/c407b52aad1c854b563c20f086c561d8d49df84d/wcpdf-mpdf.php#L128-L134

**Comment by MohamadNateqi** [comment]
@alexmigf What if you deactivate the mPDF plugin?

**Comment by alexmigf** [comment]
@MohamadNateqi that's the case, fixed.

**Comment by MohamadNateqi** [comment]
@alexmigf Looks good now.

Can you also fix the difference in the indentation?

image

**Comment by alexmigf** [comment]
@MohamadNateqi fixed, also in mPDF extension too.

**Review by MohamadNateqi — APPROVED** [review]
Nice improvements.

**Review by dwalkerpriv — APPROVED** [review]
Still good on my end 👍

**Comment by dpeyou** [comment]
@alexmigf when I switch to this PR and have IPS Pro active, the multilingual settings disappear from the '**Pro**' tab.

In the picture below, I cannot scroll down further.

**Comment by alexmigf** [comment]
@dpeyou I have updated the branch and I don't have any issues on that regard.

**Comment by dpeyou** [comment]
@alexmigf [this](https://github.com/wpovernight/woocommerce-pdf-invoices-packing-slips/pull/810#issuecomment-2196405771) has been resolved :heavy_check_mark:

**Comment by dpeyou** [comment]
@alexmigf what is this supposed to change? I have tested it on other branches as well for both free plugins.
Is it still supposed to be relevant with this PR?

enable rtl support - wcpdf

**Review by YordanSoares — APPROVED** [review]
I can't read Arabic, but it seems that it is working well 😁

image

**Comment by YordanSoares** [comment]
> But it appears that the text is wrongly placed on the document

@dpeyou did you change the template to "Simple RTL"?:

image

**Comment by dpeyou** [comment]
@YordanSoares @alexmigf  that template (Simple RTL) does not appear for me. I was under the impression that this setting was supposed to achieve that: https://github.com/wpovernight/woocommerce-pdf-invoices-packing-slips/pull/810#issuecomment-2196477234

**Comment by dpeyou** [comment]
[screen-recording(13).webm](https://github.com/wpovernight/woocommerce-pdf-invoices-packing-slips/assets/74280840/6def120b-844a-4196-bc25-66c0aa4191e2)

**Comment by alexmigf** [comment]
@YordanSoares I'm having the same issue as Darren, but even with lower versions of mPDF I can't get the Arabic font work. How did you tested it?

**Comment by alexmigf** [comment]
@YordanSoares seems to be working for me now:

Screenshot from 2024-07-08 15-59-20

**Comment by alexmigf** [comment]
@dpeyou the RTL option to work requires this branch too: https://github.com/wpovernight/woocommerce-pdf-ips-mpdf/pull/21

**Comment by dpeyou** [comment]
@alexmigf take a look at [this comment](https://github.com/wpovernight/woocommerce-pdf-invoices-packing-slips/pull/810#issuecomment-2199204584). I was already on both correct ranches. [This video ](https://github.com/wpovernight/woocommerce-pdf-invoices-packing-slips/pull/810#issuecomment-2199248377)even shows that.

I will test again, and I will test on a newer test site.

**Comment by dpeyou** [comment]
@alexmigf I tested on previous versions of mPDF and I think the issue is that i need to do `composer install`, to generate the templates in the free plugin.
However, when I do this command, this is the console output:

<pre><font color="#26A269"><b>$</b></font> composer install
<font color="#26A269">Installing dependencies from lock file (including require-dev)</font>
<font color="#26A269">Verifying lock file contents can be installed on current platform.</font>
<span style="background-color:#C01C28"><font color="#D0CFCC">Your lock file does not contain a compatible set of packages. Please run composer update.</font></span>

  Problem 1
    - dompdf/dompdf is locked to version v3.0.0 and an update of this package was not requested.
    - dompdf/dompdf v3.0.0 requires ext-dom * -&gt; it is missing from your system. Install or enable PHP&apos;s dom extension.
  Problem 2
    - masterminds/html5 is locked to version 2.9.0 and an update of this package was not requested.
    - masterminds/html5 2.9.0 requires ext-dom * -&gt; it is missing from your system. Install or enable PHP&apos;s dom extension.
  Problem 3
    - sabre/xml is locked to version 2.2.7 and an update of this package was not requested.
    - sabre/xml 2.2.7 requires ext-dom * -&gt; it is missing from your system. Install or enable PHP&apos;s dom extension.

To enable extensions, verify that they are enabled in your .ini files:
    - /etc/php/7.4/cli/php.ini
    - /etc/php/7.4/cli/conf.d/10-opcache.ini
    - /etc/php/7.4/cli/conf.d/10-pdo.ini
    - /etc/php/7.4/cli/conf.d/20-calendar.ini
    - /etc/php/7.4/cli/conf.d/20-ctype.ini
    - /etc/php/7.4/cli/conf.d/20-exif.ini
    - /etc/php/7.4/cli/conf.d/20-ffi.ini
    - /etc/php/7.4/cli/conf.d/20-fileinfo.ini
    - /etc/php/7.4/cli/conf.d/20-ftp.ini
    - /etc/php/7.4/cli/conf.d/20-gd.ini
    - /etc/php/7.4/cli/conf.d/20-gettext.ini
    - /etc/php/7.4/cli/conf.d/20-iconv.ini
    - /etc/php/7.4/cli/conf.d/20-json.ini
    - /etc/php/7.4/cli/conf.d/20-mbstring.ini
    - /etc/php/7.4/cli/conf.d/20-phar.ini
    - /etc/php/7.4/cli/conf.d/20-posix.ini
    - /etc/php/7.4/cli/conf.d/20-readline.ini
    - /etc/php/7.4/cli/conf.d/20-shmop.ini
    - /etc/php/7.4/cli/conf.d/20-sockets.ini
    - /etc/php/7.4/cli/conf.d/20-sysvmsg.ini
    - /etc/php/7.4/cli/conf.d/20-sysvsem.ini
    - /etc/php/7.4/cli/conf.d/20-sysvshm.ini
    - /etc/php/7.4/cli/conf.d/20-tokenizer.ini
You can also run `php --ini` inside terminal to see which files are used by PHP in CLI mode.</pre>

**Comment by alexmigf** [comment]
@dpeyou You don't need to run composer, the vendor directory is already built: https://github.com/wpovernight/woocommerce-pdf-ips-mpdf/tree/removes-templates

You have run composer in the wrong directory, Dompdf is from the main free plugin.

**Comment by alexmigf** [comment]
@dpeyou maybe you just need to install the missing Arabic translations. Check under the updates tab if you have any updates there. This was my issue too.

**Comment by dpeyou** [comment]
@alexmigf  Based on your explanation, I am not sure we were having the same issue (I detailed my issue with screenschots and a video). 
- Which updates are you referring to?
- Arabic script (Saudi) is already appearing correctly on my invoice and the CSS is now applied, [after this commit](https://github.com/wpovernight/woocommerce-pdf-invoices-packing-slips/pull/810/commits/b2ab3ffdb2f1a8e490031efb0ac2f60b2c6f38a9), I believe.

arabic appearing correctly after CSS fix

Previously, the Arabic text was appearing in the wrong direction, as detailed [in this comment](https://github.com/wpovernight/woocommerce-pdf-invoices-packing-slips/pull/810#pullrequestreview-2148897063).


The problem that I am facing now is that the Simple RTL template does not appear in the list of select-able templates - while in this PR, [as Yordan has it here](https://github.com/wpovernight/woocommerce-pdf-invoices-packing-slips/pull/810#issuecomment-2198794570).

**Comment by dpeyou** [comment]
@alexmigf 
Just to confirm, [this setting](https://github.com/wpovernight/woocommerce-pdf-invoices-packing-slips/pull/810#issuecomment-2196477234) is now working well for me. This is the output (with Arabic script in the correct direction):

arabic text

I am currently thinking that the issue you mentioned in our last Wednesday meeting and that [you mention here](https://github.com/wpovernight/woocommerce-pdf-invoices-packing-slips/pull/810#issuecomment-2216747652) is not related to this PR, right? The Arabic script above does not appear wihtout mPDF, so it appears to me the plugin is doing its job correctly, despite the error I mentioned in my previous comment.

**Comment by dpeyou** [comment]
I should have shared the Invoice itself set to Arabic (Saudi) instead of just demonstrating manually added Arabic script:




There are still no RTL templates in the plugin settings, as in described in Yordan's comment:
https://github.com/wpovernight/woocommerce-pdf-invoices-packing-slips/pull/810#issuecomment-2198794570

**Comment by alexmigf** [comment]
@dpeyou the mPDF templates have been removed in this [PR](https://github.com/wpovernight/woocommerce-pdf-ips-mpdf/pull/21). We are now using the default templates, which support mPDF out-of-the-box. If you need RTL, you can enable it in the General tab settings.

**Comment by dpeyou** [comment]
@alexmigf thanks for clarifying. That means this is good to go 😀

**Review by dpeyou — APPROVED** [review]
Good to go :rocket:  Does this need documenting?

**Comment by alexmigf** [comment]
@dpeyou doesn't hurt I think. Maybe a general page explaining why mPDF is an alternative engine, and for which customers is better (including the RTL setting).

**Pull request merged**